### PR TITLE
Check retries when generating report with action-junit-report

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
- Check retries when generating report with `action-junit-report`. This fixes the discrepancy in generating unit test report with the recent upgrade (by dependabot) of `mikepenz/action-junit-report`.
- Upgraded to latest v5
- Note: `steps.report.outputs.flaky_summary != '<table><tr><th>Test</th><th>Retries</th></tr></table>'` this workaround means the table for flaky test is empty.

Set the following config to `true`:
- `check_retries` to consider results of retry
- `flaky_summary` to include flaky tests in summary. However,  to generate, it requires `include_passed` to be true since other retried tests may have passed.

#### Ticket Link
none

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by toninis — APPROVED** [review]
🚀

**Comment by toninis** [comment]
Related with https://github.com/mattermost/mattermost/pull/29316

File: .github/workflows/server-test-template.yml [hunk 1]
```diff
@@ -74,15 +74,18 @@ jobs:
...
           report_paths: server/report.xml
           check_name: ${{ inputs.name }} (Results)
           job_name: ${{ inputs.name }}
           require_tests: true
+          check_retries: true
+          flaky_summary: true
+          include_passed: true
       - name: Report retried tests via webhook (master || release-*)
-        if: ${{ steps.report.outputs.retried > 0 && (github.ref_name == 'master' || startsWith(github.ref_name, 'release-')) }}
+        if: ${{ steps.report.outputs.flaky_summary != '<table><tr><th>Test</th><th>Retries</th></tr></table>' && (github.ref_name == 'master' || startsWith(github.ref_name, 'release-')) }}
```
Comment (agnivade) [inline .github/workflows/server-test-template.yml:88]: Oh wow, why is the `retried` counter removed?

**Review by agnivade — APPROVED** [review]

**Comment by agnivade** [comment]
Also, it seems like the error logs for the flaky test are buried inside the test logs and not really surfaced? Is there any way to get that without having to download and extract the zip file and locate the test failure?

File: .github/workflows/server-test-template.yml (hunk 1 above)
Comment (saturninoabril) [reply .github/workflows/server-test-template.yml:88]: It's no longer present, but I'll try to submit a PR upstream so this quick hack can be reverted.

I know @lieut-data had the changes at https://github.com/lieut-data/action-junit-report -- did those eventually make it into the upstream repo?

Comment (lieut-data) [reply .github/workflows/server-test-template.yml:89]: @saturninoabril, I regret I haven't submitted those changes upstream yet. I can try updating https://github.com/lieut-data/action-junit-report with the latest code as a stop-gap for this effort, and then submit upstream in the hopes of breaking this dependency juggling altogether?

**Comment by saturninoabril** [comment]
Merging this now to unblock other failing PRs.

**Pull request merged**